src/azul/plugins/repository/tdr_anvil/__init__.py

```diff
@@ -727,99 +733,12 @@ def convert_column(value):
-    indexed_columns_by_entity_type = {
-        'biosample': {
-            'biosample_id',
-            'anatomical_site',
-            'apriori_cell_type',
-            'biosample_type',
-            'disease',
-            'donor_age_at_collection_unit',
-            'donor_age_at_collection_lower_bound',
-            'donor_age_at_collection_upper_bound',
-        },
-        'dataset': {
-            'dataset_id',
-            'consent_group',
-            'data_use_permission',
-            'owner',
-            'principal_investigator',
-            'registered_identifier',
-            'title',
-            'data_modality'
-        },
-        'diagnosis': {
-            'diagnosis_id',
-            'disease',
-            'diagnosis_age_unit',
-            'diagnosis_age_lower_bound',
-            'diagnosis_age_upper_bound',
-            'onset_age_unit',
-            'onset_age_lower_bound',
-            'onset_age_upper_bound',
-            'phenotype',
-            'phenopacket'
-        },
-        'donor': {
-            'donor_id',
-            'organism_type',
-            'phenotypic_sex',
-            'reported_ethnicity',
-            'genetic_ancestry',
-            # Not stored in index; only retrieved to verify redundancy with
-            # biosample.part_of_dataset_id
-            'part_of_dataset_id'
-        },
-        'file': {
-            'file_id',
-            'data_modality',
-            'file_format',
-            'file_size',
-            'file_md5sum',
-            'reference_assembly',
-            'file_name',
-            'file_ref',
-            'is_supplementary',
-        },
-        'activity': {
-            'activity_id',
-            'activity_type',
-        },
-        'alignmentactivity': {
-            'alignmentactivity_id',
-            'activity_type',
-            'data_modality',
-            'reference_assembly',
-            # Not in schema
-            'date_created',
-        },
-        'assayactivity': {
-            'assayactivity_id',
-            'activity_type',
-            'assay_type',
-            'data_modality',
-            # Not in schema
-            'date_created',
-        },
-        'sequencingactivity': {
-            'sequencingactivity_id',
-            'activity_type',
-            'assay_type',
-            'data_modality',
-        },
-        'variantcallingactivity': {
-            'variatncallingactivity_id',
-            'activity_type',
-            'reference_assembly',
-            'data_modality'
-        }
-    }
+        table = one(
+            table for table in anvil_schema['tables']
+            if table['name'].removeprefix('anvil_') == entity_type
+        )
+        entity_columns = {column['name'] for column in table['columns']}
+        entity_columns.add('datarepo_row_id')
+        if entity_type in ('assayactivity', 'alignmentactivity'):
+            entity_columns.add('date_created')
```

Comment:
This deserves an explanatory comment.

Reply:
It used to be the case that  the `date_created`  column was present in BigQuery and indexed by Azul despite not being included in the schema. However, it appears that this column is no longer present in BigQuery, at least for the one dataset I looked at. 

```
select distinct column_name
from `datarepo-dev-97ad270b.ANVIL_CMG_Sample_1_20230225_ANV5_202302281509.INFORMATION_SCHEMA.COLUMNS`
where table_name = 'anvil_assayactivity' or table_name = 'anvil_alignmentactivity'
order by column_name
```

```
activity_type
alignmentactivity_id
antibody_id
assay_type
assayactivity_id
data_modality
datarepo_row_id
generated_file_id
reference_assembly
source_datarepo_row_ids
used_biosample_id
used_file_id
```

We didn't notice because there are no activities from the `assayactivity` or `alignmentactivity` tables in any bundle on `anvildev` or `anvilprod`, so we never compose a query that includes the `date_created` column. All activities are currently from the generic `activity` table (the vast majority) or the `sequencingactivity` table. 

The `date_created` field is present in the Azul API, but is always `null`. It is present because we supply default values to ensure that every document in the `activities` index has the same shape.

I suspect the proper course of action is to remove all references to `date_created` from Azul.

Reply:
Agreed at PL to remove `date_created` in a separate PR that should block this PR. The removal of the field counts as a backwards-incompatible API change.